spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastInstanceFactory.java

```diff
@@ -45,10 +41,11 @@
 	 * Create a {@link HazelcastInstanceFactory} for the specified configuration location.
 	 * @param configLocation the location of the configuration file
 	 * @throws IOException if the configuration location could not be read
+	 * @deprecated since 2.3.0 in favor of {@link HazelcastConfigFactory}
 	 */
+	@Deprecated
```

Comment:
Isn't that making `HazelcastInstanceFactory` effectively deprecated? For the reason above, I don't think we should do that and rather research how to improve the existing class to account for the customizers.